lib/ip.go

```diff
@@ -36,13 +37,56 @@ func RandIP4List(n int) []net.IP {
 	return ips
 }
 
+// RandIP4Range returns a list of randomly generated IPv4 addresses within
+// starting and ending IPrange.
+
+// note: starting and ending IPs should be in format of "1.1.1.1"
+func RandIP4Range(startIP, EndIP string) (net.IP, error) {
+	StartIPRaw := net.ParseIP(startIP).To4()
+	if len(StartIPRaw) == 0 || len(StartIPRaw) > net.IPv4len {
+		return nil, fmt.Errorf("range is Invalid")
+	}
+	StartIPInt := binary.BigEndian.Uint32(StartIPRaw.To4())
+	EndIPRaw := net.ParseIP(EndIP).To4()
+	if len(EndIPRaw) == 0 || len(EndIPRaw) > net.IPv4len {
+		return nil, fmt.Errorf("range is Invalid")
+	}
+	EndIPInt := binary.BigEndian.Uint32(EndIPRaw.To4())
+	if StartIPInt > EndIPInt {
+		return nil, fmt.Errorf("range is Invalid")
+	}
+	rangeIP := binary.BigEndian.Uint32(RandIP4().To4())
+	temp := EndIPInt - StartIPInt
+	if temp <= 0 {
```

Comment:
Since we know that `StartIPInt <= EndIPInt`, this can only yield `>= 0`, but not less than 0, so it can be simplified to:

```suggestion
	if temp == 0 {
```